src/java/picard/sam/MergeSamFiles.java

```diff
@@ -83,6 +90,9 @@
     @Option(doc = "Comment(s) to include in the merged output file's header.", optional = true, shortName = "CO")
     public List<String> COMMENT = new ArrayList<String>();
 
+    @Option(shortName = "INTERVALS", doc = "An interval list file that contains the locations of the positions to merge. Assume bam are sorted and indexed.", optional = true)
```

Comment:
What I meant was that one could think about scatter/gathering the merge
process by creating a bunch of processes that do merge on non-overlapping
intervals, and then concatenate all the resulting bams together (not sure
if it's worth it, but one could try...) the problem is that then the reads
that over lap two interval-lists will get included twice in the
concatenation step...

What are you starting with that you want this feature? (sorry to be nosy,
but I'm trying to understand the use-case.)

On Tue, Oct 27, 2015 at 3:59 PM, Pierre Lindenbaum <notifications@github.com
> wrote:

> In src/java/picard/sam/MergeSamFiles.java
> <https://github.com/broadinstitute/picard/pull/314#discussion_r43177631>:
>
> > @@ -83,6 +90,9 @@
> >      @Option(doc = "Comment(s) to include in the merged output file's header.", optional = true, shortName = "CO")
> >      public List<String> COMMENT = new ArrayList<String>();
> >
> > +    @Option(shortName = "INTERVALS", doc = "An interval list file that contains the locations of the positions to merge. Assume bam are sorted and indexed.", optional = true)
>
> My initial idea was to quickly create a merged bam for a given (set of)
> regions before calling SNPs
>
> and the resulting mess that would arise from that
>
> Doesn't MergingSamRecordIterator handle this problem ? if not, you're
> absolutely right, that would be a problem. :-(
>
> feel free to close that pull request if this is the case.
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/broadinstitute/picard/pull/314/files#r43177631>.
>